### ISSUE TITLE
Darktable 3.6.1 Crashes when selecting another image in darkroom

### ISSUE THREAD

**Issue by andy-hnq** [darktable-org/darktable#10384]
**Introduction**

**Describe the bug/issue**

**To Reproduce**

This seems to be intermittent but I've had several crashes over the past few days. I've had it when switching to a new (.CR2) image in darkroom mode, both when using the thumbnail strip at the bottom and when using the spacebar to move to the next image.

**Expected behavior**

Should move to the selected image without crashing

**Which commit introduced the error**

Installed 3.6.1 using darktable-3.6.1-win64.exe downloaded from darktable.org on 2nd Nov. Prior install was 3.4 which was uninstalled using control panel before installing 3.6.1

**Platform**

* darktable version : 3.6.1
* OS                : Windows 10.0.14393
* Memory            : 12Gb
* DrMingw 0.9.2

**Additional context**

 - I cannot reproduce the issue on demand unfortunately.
 - Are the steps above reproducible with a fresh edit (i.e. after discarding history)? - the images involved were captured after the upgrade but the default module stack had been modified - certainly the Crop and Rotate module was in use, possibly exposure with a drawn mask.
 - Do you use lua scripts?
   - HDRMerge shows as started but I have not used it since upgrading
   - 
[darktable_bt_TTJ7B1.txt](https://github.com/darktable-org/darktable/files/7516421/darktable_bt_TTJ7B1.txt)
[darktable_bt_QRMJC1.txt](https://github.com/darktable-org/darktable/files/7516422/darktable_bt_QRMJC1.txt)
[darktable_bt_H54IC1.txt](https://github.com/darktable-org/darktable/files/7516423/darktable_bt_H54IC1.txt)

**Comment by MStraeten** [comment]
is this reproducible in an fresh darktable environemt (cleaned \AppData\Local\darktable - don't forget to backup first - and also cache directory \AppData\Local\Microsoft\Windows\INetCache\darktable)?
can you provide the cr2 (at least camera model name to be able to use public cr2 files) and xmp files?

**Comment by andy-hnq** [comment]
Thanks for the reply.

At the moment the issue is not reproducible even without any cleanup. It may be that my efforts to repeat what I was doing (both with the same images and with new ones) were missing something. I should be capturing and processing some similar images this week so I will see if it recurs and if so try to figure out a more precise set of steps and upload the relevant files.

The camera is a Canon EOS 70D.

**Comment by andy-hnq** [comment]
Hardly 'reproduced' but I have spent several hours processing some TIF colour negative scans and had one more crash towards the end of the session. The better images had been given two stars and I had the film roll filtered to two stars (the same was likely true for the crashes on the 9th). I had the first image open and clicked the thumbnail strip to select the fifth. Unfortunately by that stage in the session the edit position was far from simple. All of the images had been edited plenty with negadoctor, crop (some crop and rotate), orientation for some, colour balance rgb (some with multiple instances), exposure for some, filmic rgb for some, haze removal for some and retouch for one to remove a dust spot.

DT did seem to start struggling in some cases to update the image when tweaking settings, for example in colour balance rgb before it crashed. This may or may not be relevant.

So this time TIF instead of CR2 and negative scans instead of DSLR images.

I started DT again and tried to repeat it without success.

[darktable_bt_6N63C1.txt](https://github.com/darktable-org/darktable/files/7541969/darktable_bt_6N63C1.txt)

**Comment by eternalinflation** [comment]
I can reproduce this fairly consistently with DT up to v3.6.1 on Gentoo (portage up to date) since a couple of days, so may be related to a recently updated dependency?

Steps to reproduce (regardless whether fresh profile or not):
1. Lighttable: double click random image to open in Darkroom
2. Go back to Lighttable and double click another random image
3. DT crashes with Segmentation fault

I can not reproduce this with DT built from master as of yesterday, though. I haven't tracked the issue down to a certain commit but would be willing to check with some guidance.

**Comment by andy-hnq** [comment]
Thanks for testing. Do you get a backtrace that looks like mine? 

I've had no further crashes since my previous post. Nothing has changed AFAIK.

**Comment by AlicVB** [comment]
reading your backtraces, this seems to comes from modulegroups module. But sadly those windows backtraces are not really helpful...

@eternalinflation : to be sure I understand you right, can you confirm that you have the issue with 3.6.1 but not with current master ?
If it's the case, then I guess that one of the recent changes in modulegroup has fixed the issue, and then upcoming 3.8 will be ok...

**Comment by eternalinflation** [comment]
@AlicVB It turned out that _any_ build from Git works OK incl. v3.6.1, so in my case the issue seems at least related to the [ebuild](https://gitweb.gentoo.org/repo/gentoo.git/tree/media-gfx/darktable/darktable-3.6.1-r2.ebuild), i.e. the package configuration, and the 'fix' is likely due to the build environment rather than a certain commit.
I'm attaching logs as a diff between the output from the direct Git build (left) and the package/portage build (right):
[build-diffs.log](https://github.com/darktable-org/darktable/files/7575971/build-diffs.log), hopefully it provides some insight.
Even if this issue may be related to the build environment I'd think it'd still be worth looking at what's causing a segfault. 
Let me know what else I could test / provide.

**Comment by andy-hnq** [comment]
Another crash - once again I hit the spacebar to move to the next image in darkroom. I'd been using DT for literally hours before this and had done the same thing many times without a crash.

This one is in libgtk... possibly a different issue, but like I said, the same action on my part.

One possible common factor - I had done a selective copy/selective paste on several modules to replicate colour balance rgb (two instances, one with a paramateric mask), filmic rgb (off on this occasion), exposure and was then tweaking each image individually. I had changed the tint in the WB module on the image just before the crash.

I tried to repeat it with the same modules copied from the same surce image onto some other target images after restarting but DT did not crash.

[darktable_bt_X8M9C1.txt](https://github.com/darktable-org/darktable/files/7584747/darktable_bt_X8M9C1.txt)

**Comment by MStraeten** [comment]
Please provide a xmp with a scenario that crashes. Without no one is able to reproduce or get an idea what could be causal.
Are you using plain standard raw files?

**Comment by andy-hnq** [comment]
Hi, I completely understand the need to reproduce this but the first step in doing that is for me to be able to repeat the issue on my own machine. I have not been able to do that - apart from the three crashes in quick succession on the 9th Nov this has proved to be quite rare. I will keep striving to figure out a set of repeatable factors but with so few crashes it is difficult.

Four of the five crashes (three on the 9th and one yesterday) have been with CR2 raws from my Canon EOS 70D. One (Nov 15th) was with a TIFF produced by VueScan from a negative scanner.

**Comment by eternalinflation** [comment]
I think I can reproducibly nail it down to the following:
````Thread 83 "worker res 0" received signal SIGSEGV, Segmentation fault.
[Switching to Thread 0x7fff9d441640 (LWP 235933)]
0x00007fffcd1aa077 in pixel_correction (exposure=<optimized out>, factors=factors@entry=0x55555812f3f0, sigma=sigma@entry=1.41421354) at /usr/src/debug/media-gfx/darktable-3.6.1-r2/darktable-3.6.1/src/iop/toneequal.c:783
783         result += gaussian_func(expo - centers_ops[i], gauss_denom) * factors[i];
(gdb) ````

**Comment by eternalinflation** [comment]
Given
````
#ifdef _OPENMP
#pragma omp simd aligned(centers_ops, factors:64) safelen(PIXEL_CHAN) reduction(+:result)
#endif
````
right before the line that triggers the SIGSEGV I rebuilt DT without _OPENMP - and the problem is gone for me.

CPU is:
````
processor       : 3
vendor_id       : GenuineIntel
cpu family      : 6
model           : 94
model name      : Intel(R) Xeon(R) CPU E3-1225 v5 @ 3.30GHz
stepping        : 3
microcode       : 0xea
````

**Comment by andy-hnq** [comment]
Hmm... I thought I had reproduced it. I managed 3 libgtk crashes in quick succession on the same image and was trying to figure out a summmarised set of steps and then... the crashes stopped.

I had decided to try a fresh profle and cache directory on the off-chance that this might make things worse instead of better.. .and it seemed to do exactly that but I could not then repeat the same result.

What I did observe was that the history stack in the database (and UI) seems to be getting out of sync with the XMP for some images. I'm not certain how this happens but I have observed it a few times. It _may_ be related to going backwards in the history stack and then applying new changes to replace those that I did not like. 

I'm not sure whether the history sync symptom is related to the crashes but I am sure that I saw this issue on the image that triggered the three crashes today. With the benefit of hindsight what I should have done was to take a copy of the XMP files straight after the crash.

I do think that my fresh profile and cache prove that the problem is not being caused by some fossilised stuff haging around from an older DT version.

**Comment by andy-hnq** [comment]
I was looking for any issues related to XMP/History and found this one. Sounds pretty similar

https://github.com/darktable-org/darktable/issues/9794

**Comment by TurboGit** [comment]
Indeed, very similar. This happens into the modulegroups lib! Looks like we have some kind of memory corruption there. Would be nice to be able to reproduce!!!!

**Comment by AlicVB** [comment]
yes I've spotted modulegroups too. but what is surprising and I can't understand is that according to @eternalinflation test, this doesn't happen when he compile *any* git versions. iiuc, that would point to some compilation parameters that are not handled well by dt...
but in modulegroups, iirc, there's absolutely nothing fancy in term of computation or whatever... no openmp, etc...

second problem, is that -of course- I've not managed to reproduce this issue, so I'll have a look at modulegroup code, but don't expect too much...

**Comment by andy-hnq** [comment]
I think there may be multiple issues here, for certain I get backtraces both in modulegroups and in libgtk. Both happen when selecting a new image inside darkroom. Are they the same or different? I don't know. The same question can be asked of the Linux issues - one of these happens in lighttable rather than darkroom. I'm not suggesting that we raise multiple issues at this time - I don't think we know enough yet but we should keep an open mind and not _assume_ that we are all talking about the same problem.

I've been trying to reproduce the XMP/database mismatch but so far without success. It definitely happened - several of my images showed up much brighter and some were majorly clipping when I imported them into the clean DT profile... which should not have happened if the XMP was correctly reflecting what I'd been doing before I reset the profile.

**Comment by eternalinflation** [comment]
Although my issue may have a different reason I thought I'd still share my findings:

I built my (crashing) binaries with the `--fomit-frame-pointer --ftree-vectorize` gcc flags, especially the latter quite possibly doesn't play nice with the _OPENMP option enabled (see SIMD pragma and possibly resulting multiple attempts at vectorization). Removing that gcc flag generates a working binary also with _OPENMP enabled. 
The build script from Git is using its own, sane gcc options, which is why the direct Git built works as well.

I hope this saves time for anyone else who's running into the same issue.

**Comment by AlicVB** [comment]
you're right, I think that there's multiple issues.
hopefully one of them is now understood (fixed ?) reading last @eternalinflation msg.
So there's now the "modulegroup" one (if we consider they are not the same.
For this one, are you able to test last dev version ? I know you are on windows, but even if you are not in position to compile by yourself, you may grab a very recent development version on pixl.us forum...
Thanks

**Comment by TurboGit** [comment]
@AlicVB : I don't have time for a PR tonight, but will do tomorrow morning if not don't beat me on it... This looks wrong:

in `_dt_dev_image_changed_callback()`:

```
  if(sqlite3_step(stmt) == SQLITE_ROW)
  {
    const char *preset = (char *)sqlite3_column_blob(stmt, 0);
    dt_lib_presets_apply(preset, self->plugin_name, self->version());
  }
```

The preset is not a blob but the a name as per SELECT above.

**Comment by TurboGit** [comment]
@AlicVB : I have fixed this part this morning.

**Comment by TurboGit** [comment]
@andy-hnq : What would help a lot here is to be able to get a backtrace from GDB. Sadly I cannot help on Windows.

**Comment by andy-hnq** [comment]
> @andy-hnq : What would help a lot here is to be able to get a backtrace from GDB. Sadly I cannot help on Windows.

If anyone knows how to do a darktable postmortem (on Windows) then I'd very much like to know. My C/C++ skills are very rusty indeed and I no longer have tools like Visual Studio available.

**Comment by andy-hnq** [comment]
Following up on the DB/XMP mismatch. I queried the sqlite DB and wrote some perl to find any remaining mismatches. The ones that I found yesterday have been resolved by my attempts to reproduce the crash but there were still six remaining. In all cases the XMP had fewer steps than the DB/UI.

\2021-06/IMG_5412.CR2.xmp db ops = 17, xmp ops = 16     <== missing monochrome(off) at end. monochrome(on) is not in stack
\2021_05_30/IMG_5335.CR2.xmp db ops = 27, xmp ops = 0   <== xmp file is all nulls (!)
\2021_07_20/IMG_5704.CR2.xmp db ops = 24, xmp ops = 23  <== missing sharpen(off) at end. sharpen(on) is not in stack
\2021_08_02/IMG_5848.CR2.xmp db ops = 15, xmp ops = 14  <== missing filmic at end - prev step in UI is also filmic, both are enabled
\2021_11_16/IMG_6408.CR2.xmp db ops = 20, xmp ops = 19  <== missing WB at end, prev step in UI is WB, both enabled
\2018-02/IMG_3335.CR2.xmp db ops = 21, xmp ops = 15     <== missing default modules at beginning. 

I don't know if this is related to the crash but it was definitely impacting the image that caused my crashes yesterday. I'd also say there is a good chance that it impacted the other images that have crashed. On the other hand, I have not so far had a crash with any of the remaining images.

Some thoughts...
- some of these images have not been touched since before I upgraded so the mismatch happened with DT 3.4
- others are new images since the upgrade so mismatches also happen with 3.6.1
- I can reproduce the final one - do a selective copy/paste of some history steps from another image in overwrite mode. This was not the type of mismatch involved in the crashes though.
- I'm still trying to reproduce the 'missing last step' issue.
- xmp all nulls... I have no idea.

**Comment by eternalinflation** [comment]
I just realized that Github did not generate a notification after I edited an earlier comment of mine https://github.com/darktable-org/darktable/issues/10384#issuecomment-977125968. For everybody who's interested please re-read the comment. 
Just to clarify my issue is triggered by switching between lighttable and darkroom as mentioned in [this comment](https://github.com/darktable-org/darktable/issues/10384#issuecomment-974644848). I wasn't able to reproduce when staying in darkroom using either space bar or the film strip.

**Comment by andy-hnq** [comment]
I was unable to find anything like a nightly Win32 build so I followed the instructions at https://github.com/darktable-org/darktable/blob/master/packaging/windows/BUILD.md and built my own installer from git.

I got the build to run after a few issues and have just installed it. I will report back if any further crashes happen.

A question... back when I got paid to write C/C++ we used a debug memory allocator that overwrote memory with known junk values just after allocation and just before free. It was pretty good at improving our chances to pick up bugs like using unititialised values or accessing memory after it had been freed. Is this possible with a (custom!) darktable build?

**Comment by TurboGit** [comment]
>  Is this possible with a (custom!) darktable build?

You can build with libasan, see:

`$ ./build.sh --help`

**Comment by dterrahe** [comment]
Disclaimer: I haven't tried to reproduce or even read the backtraces provided (honestly, I'm getting the impression that two separate bugs are being discussed at the same time here?).

But just reading about modulegroups and changing images in darkroom reminded me of similar bugs where switching from an image with several instances of the same module to an image that only had one (the base) instance would cause the now deleted extra instances to still be referenced. This could be happening with widgets in QAP for example. If this were the cause, it might be somewhat easier to identify a reproducing set of xmps.

**Comment by andy-hnq** [comment]
> But just reading about modulegroups and changing images in darkroom reminded me of similar bugs where switching from an image with several instances of the same module to an image that only had one (the base) instance would cause the now deleted extra instances to still be referenced. This could be happening with widgets in QAP for example. If this were the cause, it might be somewhat easier to identify a reproducing set of xmps.

That is certainly true of the images involved in my crashes. In one case multiple instances of exposure, in another I have multiple colour balance rgbs. The second (and third sometimes) instance uses a mask. You may be correct but unfortunately it's still not reproducible - I've been back to those images several times trying to reproduce the issue but without success.

**Comment by eternalinflation** [comment]
Agreed - I didn't mean to clutter this discussion with a different topic. I created https://github.com/darktable-org/darktable/issues/10498 instead.

**Comment by andy-hnq** [comment]
I have a reproducible issue. It isn't exactly the original issue but I can reproduce it on an otherwise unmodified image (i.e. statting from just the default modules). 

This was using my self built version from Friday's git repository... which does contain Pascal's fix. The installer name is 'darktable-3.7.0+1549~g5de65d1fe-win64'

It produces a libgtk access violation on address -1

- Open the image in darkroom
- switch on color balance RGB and select 'add basic colorfulness' preset
- Go back to lighttable
- Open the image again in darkroom
- add a second color balance RGB using 'new instance' and select 'add basic colorfulness' preset
- disable the new module
- enable the new module
- delete the new module
- select lightroom... crash.

This does not produce the crash if I select another image in darkroom so that issue is still not reproduced.

Should I raise this as another issue or keep the discussion here?

Do the steps above produce a crash for other people?

**Comment by dterrahe** [comment]
I can reproduce _a_ crash using simplified steps:
- select an image in lighttable
- press 'd' to switch to darkroom
- create new instance of any module
- toggle instance off and back on by pressing its first header icon ("enable")
- delete instance
- press 'd' to switch to lighttable

I get an "g_object_ref: assertion 'G_IS_OBJECT (object)' failed" in `gtk_widget_grab_focus(dt_ui_center(darktable.gui->ui));`

It looks like possibly the _enable_ toggle button still had the focus even after being deleted and then when gtk tries to switch focus to the central ui, it tries to access the deleted button.

Because if I don't follow these exact steps, for example, I switch to lighttable by double clicking the image instead of pressing the shortcut, there is no crash. In this case, the center already has the focus.

Also, if I create a shortcut to the enable button of the module and use _that_ to toggle it off and on before deleting it, there is no crash.

This may be a completely different bug; what was the error your crash log reported?

**Comment by AlicVB** [comment]
@andy-hnq @dterrahe : I can reproduce an issue with your 2 steps by steps and I can confirm that you both hit the same problem. I'll try to have a look...

**Comment by andy-hnq** [comment]
Sounds like the same issue to me... along with some analysis of the reason. After I'd posted my message I also managed to reproduce it using a second instance of exposure (in addition to the instance created when the image is imported) so the first few steps are not essential and neither is the specific module (color balance rgb).

Sample backtrace attached - sorry should have done this with the original post.

[darktable_bt_AO1QD1.txt](https://github.com/darktable-org/darktable/files/7620190/darktable_bt_AO1QD1.txt)

**Comment by andy-hnq** [comment]
I just had another look at my previous backtraces. I suspect from that the issue reproduced today is a different problem.

All of the other backtraces include _display_module_trouble_message_callback ... both the libmodulegroups and libgtk crashes.

What is a 'module trouble message' please?

Would selecting a new image always call this function or is it only called if there is some 'trouble' with the module?

**Comment by andy-hnq** [comment]
Partially answering my own question I went looking for other issues mentioning this function and found a reference to dt_iop_set_module_trouble_message.

Using grep on the source revealed...

```
$ find . -type f |xargs grep -l dt_iop_set_module_trouble_message
./common/imagebuf.c
./develop/imageop.c
./develop/imageop.h
./develop/pixelpipe_hb.c
./iop/cacorrect.c
./iop/channelmixerrgb.c
./iop/lens.cc
./iop/temperature.c
```

I think that white balance (aka temperature) is the only one I was using on the images that triggered the crash. I've seen a warning banner in this module when messing with multiple illuminants but that was definitely not happening when the crashes occurred and did not happen on the images involved, wither beforehand or afterwards.

As far as I am aware there were no banner messages in **any** modules when the crashes occurred... and none beforehand and none afterwards for those images.

I wondered... if it wasn't one of those modules then maybe...

./develop/pixelpipe_hb.c

```
      if(piece->enabled != hist->enabled)
      {
        if(piece->enabled)
          dt_iop_set_module_trouble_message(piece->module, _("enabled as required"), _("history had module disabled but it is required for this type of image.\nlikely introduced by applying a preset, style or history copy&paste"), NULL);
        else
          dt_iop_set_module_trouble_message(piece->module, _("disabled as not appropriate"), _("history had module enabled but it is not allowed for this type of image.\nlikely introduced by applying a preset, style or history copy&paste"), NULL);
        dt_print(DT_DEBUG_PARAMS, "[pixelpipe_synch] enabling mismatch for module %s in image %i\n", piece->module->op, imgid);
      }

```

History copy/paste was definitely happening. I don't know how to cause either of the trouble conditions above so not sure how to test the theory.

**Comment by AlicVB** [comment]
@dterrahe : don't bother with a  new issue, I have the fix almost ready (just need some tests to be sure that don't break anything ;)

**Comment by dterrahe** [comment]
> @dterrahe : don't bother with a new issue, I have the fix almost ready (just need some tests to be sure that don't break anything ;)

Sorry, I meant I was submitting a _fix_ (not a bug). Done now. But maybe you have a better/more fundamental solution. In which case you probably also removed the old workaround (without, as I did, explicitly assigning focus).

**Comment by andy-hnq** [comment]
Hmm... I built again from source just now to pick up the fix to the focus issue.

...another crash I'm afraid :(

I was surprised that my installer had exactly the same name but 'git pull' had definitely done its stuff (I checked the source and the changes are there) and the build had... well... built.

So to make sure I tried the previously crashing (simpler) steps.

- open an image in darkroom
- select 'new instance' on exposure
- move the EV slider on the new instance
- disable the new instance
- enable the new instance
- delete the new instance... crash

This crashes sooner and has a different looking backtrace but I guess that the change has an issue? ... or could it be my build?

Tried twice with the same outcome.

[darktable_bt_Z8XRD1.txt](https://github.com/darktable-org/darktable/files/7628343/darktable_bt_Z8XRD1.txt)
[darktable_bt_LB6PD1.txt](https://github.com/darktable-org/darktable/files/7628341/darktable_bt_LB6PD1.txt)

If I use 'duplicate' and don;t touch the slider then it doesn't crash, even when going back to lighttable

I then started to doubt whether I was moving the slider yesterday or not but I think I was.

So I tried the exact original steps using color balance rgb from my post. These steps also crash on delete... without me touching any sliders etc in the module, just selecting the preset.

[darktable_bt_LB6PD1.txt](https://github.com/darktable-org/darktable/files/7628391/darktable_bt_LB6PD1.txt)

**Comment by dterrahe** [comment]
With current master seems I can't reproduce using your steps.

**Comment by AlicVB** [comment]
I can't reproduce too with current master (after @dterrahe commits)...
Can you by any chance try to compile with only the first fix ?
Something like : `git checkout c359de81de9d1a6bdc6b605850ed891e562ab577` + all your compile steps

Thanks !

**Comment by andy-hnq** [comment]
OK so (before @AlicVB's post) I thought I would try a clean build. The build notes don't say how to do this so in the end I wiped the build directory completely and started from the cmake -G "MSYS Makefiles" ... step

I also did another git pull - it said I was up to date.

The build took about an hour so was definitely doing more.

Uninstalled and then installed the new one.

Same issue still :(

Another thought (on repeatability)... The build notes say

_NOTE: The package created will be optimized for the machine on which it has been built, but it could not run on other PCs with different hardware or different Windows version. If you want to create a "generic" package, change the first cmake command line as follows:_

Not sure how it is optimised for my machine or whether this might make a difference. Should I do the 'generic' thing or do the optimised thing? I guess keep it optimised otherwise I will be changing two things at once.

It's getting late for me now. I will try the checkout and another build tomorrow.

**Comment by andy-hnq** [comment]
Build done from 'git checkout c359de81de9d1a6bdc6b605850ed891e562ab577'.

No crashes from my original steps nor on delete :)

I used the original build steps, so still 'optimised' (whatever that means).

**Comment by dterrahe** [comment]
Thanks for all your work trying to reproduce the crash. It can be very tedious but is very helpful. The devs can't catch all bugs themselves (they tend to not show up in the workflows they've been tested on during development because then they would already have been fixed) so volunteers taking the trouble to file bugs are essential. Thanks again!

**Comment by AlicVB** [comment]
@dterrahe : so the simplest fix is to revert your second commit, imho... what do you think ?
this one : https://github.com/darktable-org/darktable/commit/9d8d38674e1b57b9b741e07e260526679f3e3e7f

**Comment by dterrahe** [comment]
> what do you think ?

I got the impression from the latest message from @andy-hnq that he no longer is seeing the problem, so nothing to fix. Did I misread that? Do you still see crashes yourself?

**Comment by AlicVB** [comment]
well, I may very well be wrong , but I've read : 
> Build done from 'git checkout c359de8'.
No crashes from my original steps nor on delete :)

So for me dt compiled at this commit doesn't crash, contrary to master... @andy-hnq : who is right ? :)

And no, I don't see any crash or suspect msg here with master...

**Comment by andy-hnq** [comment]
The crashes only stopped by going back to that specific commit.

No idea why I see them and you folks don't. Sorry.

**Comment by dterrahe** [comment]
Ah, sorry, yes I _did_ misread/not check which commit was being tested. So yes, reverting the second part would be the easy "fix" for now. @TurboGit you may want to do this now so we get more testing on that.

Still strange that it causes crashes for him but not us. Maybe a gtk version issue.

**Comment by andy-hnq** [comment]
> Maybe a gtk version issue.

I thought that GTK comes from the Darktable build on Win32. 

Certainly I see it in a subdirectory under 'darktable/build'.

The backtrace (yesterday) says 'libgtk-3-0.dll	3.24.29.0'

... but if I do file/properties on the one in today's build directory I see '3.24.30.0'

I also kept yesterday's (second) master build - this has the same version.

If I use procexp it says that darktable is using the one in Program Files\darktable\bin ... which is '3.24.30.0'

I've also checked the PATH to see if there are any others but found none. I do have gimp and Inkscape which both have libgtk 2 versions but I dont think these are interfering.

I am puzzled by this!

**Comment by TurboGit** [comment]
Ok, commit reverted. Please double check that master is ok. Thanks!

**Comment by andy-hnq** [comment]
My bad. Obvious error... looking at the wrong backtrace file.

The correct backtrace agrees... libgtk-3-0.dll	3.24.30.0

**Comment by andy-hnq** [comment]
I checked out master and then left another clean build running while doing some other stuff. When I came back to it...

[ 55%] Linking C shared library libdarktable.dll
C:/msys64/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/11.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: CMakeFiles/lib_darktable.dir/objects.a(imageio_libraw.c.obj): in function `dt_imageio_open_libraw':
C:/msys64/home/andy/darktable/src/common/imageio_libraw.c:226: undefined reference to `libraw_open_wfile'
collect2.exe: error: ld returned 1 exit status
make[2]: *** [bin/CMakeFiles/lib_darktable.dir/build.make:3322: bin/libdarktable.dll] Error 1
make[1]: *** [CMakeFiles/Makefile2:2227: bin/CMakeFiles/lib_darktable.dir/all] Error 2
make: *** [Makefile:156: all] Error 2

Will look some more tomorrow.

**Comment by dterrahe** [comment]
When you do a checkout, do you also run `git submodule update --init`?

**Comment by andy-hnq** [comment]
> When you do a checkout, do you also run `git submodule update --init`

No I didn't. Thanks - that fixed the build issue.

FYI this was the output...
```
$ git submodule update --init
Submodule path 'src/external/LibRaw': checked out '30595a731f3bea78f0410426b73ef3afbd3f64c7'
Submodule path 'src/tests/integration': checked out '4b929fa87b7d8d8a7f99fb0b0905800e2e602a85'
```
I tried the previously crashing steps on the new build and saw no issues.

**Comment by dterrahe** [comment]
Could you also please check on a version _before_ 9d8d386 was reverted? Maybe it was innocent and can be included again (rereverted)?

**Comment by andy-hnq** [comment]
Will do but can you tell me the right git steps to get the relevant version pls?

**Comment by TurboGit** [comment]
@andy-hnq : Can you try this commit:

```
$ git checkout c95cd44f20d8d39c8ea039f3023f28105646ba8e
$ ./build.sh
```

Install and try as usual. TIA.

**Comment by andy-hnq** [comment]
I built from that commit. It still crashes when deleting the second module instance.

I note that Pascal's instructions say 'build.sh'... I am using the instructions at https://github.com/darktable-org/darktable/blob/master/packaging/windows/BUILD.md and using cmake. Are some of you doing win32 builds with cmake to build an installer and then installing that?

**Comment by alex6480** [comment]
I get the same problem on my computer. It happens quite frequently, especially in complex images that are are slow to process.
It only crashes when switching images from one to another. The issue has gotten progressively worse. It used to rarely crash, but now it crashes almost every time I switch image.

[darktable_bt_1NLZH1.txt](https://github.com/darktable-org/darktable/files/8185510/darktable_bt_1NLZH1.txt)
[darktable_bt_9GY0H1.txt](https://github.com/darktable-org/darktable/files/8185511/darktable_bt_9GY0H1.txt)
[darktable_bt_7L6YH1.txt](https://github.com/darktable-org/darktable/files/8185512/darktable_bt_7L6YH1.txt)
[darktable_bt_FI1AI1.txt](https://github.com/darktable-org/darktable/files/8185513/darktable_bt_FI1AI1.txt)
[darktable_bt_0DS4H1.txt](https://github.com/darktable-org/darktable/files/8185514/darktable_bt_0DS4H1.txt)
[darktable_bt_AGPRI1.txt](https://github.com/darktable-org/darktable/files/8185515/darktable_bt_AGPRI1.txt)
[darktable_bt_H54IC1.txt](https://github.com/darktable-org/darktable/files/8185516/darktable_bt_H54IC1.txt)
[darktable_bt_YG2PI1.txt](https://github.com/darktable-org/darktable/files/8185517/darktable_bt_YG2PI1.txt)
[darktable_bt_9ZDRI1.txt](https://github.com/darktable-org/darktable/files/8185518/darktable_bt_9ZDRI1.txt)
[darktable_bt_4AAVI1.txt](https://github.com/darktable-org/darktable/files/8185519/darktable_bt_4AAVI1.txt)
[darktable_bt_WXD1I1.txt](https://github.com/darktable-org/darktable/files/8185520/darktable_bt_WXD1I1.txt)
[darktable_bt_8GYMI1.txt](https://github.com/darktable-org/darktable/files/8185521/darktable_bt_8GYMI1.txt)
[darktable_bt_5NVWI1.txt](https://github.com/darktable-org/darktable/files/8185522/darktable_bt_5NVWI1.txt)
[darktable_bt_TQG1I1.txt](https://github.com/darktable-org/darktable/files/8185523/darktable_bt_TQG1I1.txt)
[darktable_bt_9MS1I1.txt](https://github.com/darktable-org/darktable/files/8185524/darktable_bt_9MS1I1.txt)

I can send some images if you tell me where to send them to, but it doesn't seem to be dependent on the image I edit. I only edit Sony RAWs (.ARW)

**Comment by TurboGit** [comment]
@alex6480 : We are not interested about 3.8.0 report since 3.8.1 has been released with many fixes. Please try with latest stable version.

**Comment by alex6480** [comment]
Thanks for the information. I installed the newest version.

The bug is still reproducible. I think it happens when switching from one image to the next while it is still processing a change on the first. If I don't edit the image, I can switch around without crashes.

[darktable_bt_MG5FI1.txt](https://github.com/darktable-org/darktable/files/8185651/darktable_bt_MG5FI1.txt)

**Comment by gi-man** [comment]
@alex6480 I think you will get better support if you post the darktable log instead of the windows error file. Run darktable from cmd with the tags -d opencl -d perf  and upload the darktable log. The log is in this windows folder.  Start a new thread with your system info per the Issue template.

https://www.darktable.org/about/faq/#faq-windows-logs
I read a lot of information in the manual to turn on some debug settings, but I cannot see any debug information. Where can I find those debug logs?
The Windows version of dt by default logs its debug information to the following places:

Windows 10: C:\Users\[username]\AppData\Local\Microsoft\Windows\INetCache\darktable\darktable-log.txt

**Comment by alex6480** [comment]
@gi-man Thanks for the suggestion. I couldn't reproduce the issue today, maybe it's because I restarted my computer in-between. I'll make sure to grab a debug log, if it happens again.

**Comment by alasta4204** [comment]
Good day. I am getting similar behavior on ubuntu 20.04 Darktable 3.8.1 What happens is that when you hit delete to get rid of a dud image it doesn't work. If you try again, fairly quickly the whole PC freezes as in this screenshot. Clicking delete on the mouse works fine ( that's how I got this screenshot ) Ubuntu is still running according to my Conky, Alt F2 to restart gnome doesn't work and I can't get a terminal with Cntrl alt t or Cntrl alt  f1. Cntrl  alt f7 takes me to a login window which allows a restart 

![Darktable 3 8 1 Freeze of UI](https://user-images.githubusercontent.com/44087152/164968044-e734cd1c-247c-42e0-b579-0a58d12ef913.png)

**Comment by TurboGit** [comment]
Should be fixed in #12125.
